source/include/FreeRTOS_IP.h

```diff
@@ -44,11 +44,15 @@
 
 /* Constants defining the current version of the FreeRTOS+TCP
  * network stack. */
-#define ipFR_TCP_VERSION_NUMBER    "V2.3.999"
-#define ipFR_TCP_VERSION_MAJOR     2
-#define ipFR_TCP_VERSION_MINOR     3
+#define ipFR_TCP_VERSION_MAJOR    2
+#define ipFR_TCP_VERSION_MINOR    3
 /* Development builds are always version 999. */
-#define ipFR_TCP_VERSION_BUILD     999
+#define ipFR_TCP_VERSION_BUILD    999
+#define __VERSION( v1, v2, v3 )    "V" # v1 "." # v2 "." # v3
+#define _VERSION( v1, v2, v3 )     __VERSION( v1, v2, v3 )
```

Comment:
MISRA doesn't like the use of `#` operator. Ref: Misra c-2012 rule 20.10 (advisory).

Also, these version numbers are updated together - and only when a release happens. Not completely sold on the value added by using the above expression to generate the `ipFR_TCP_VERSION_NUMBER` string.

Reply:
I have cancelled the change to the macro `ipFR_TCP_VERSION_NUMBER`.